brainiak/utils/fmrisim.py

```diff
@@ -1430,14 +1457,46 @@ def _generate_noise_temporal_drift(trs,
 
     """
 
-    # Calculate the cycles of the drift for a given function.
-    cycles = trs * tr_duration / period
+    # Calculate drift differently depending on the basis function
+    if basis == 'discrete_cos':
+
+        # Specify each tr in terms of its phase with the given period
+        timepoints = np.linspace(0, trs - 1, trs)
+        timepoints = ((timepoints * tr_duration) / period) * 2 * np.pi
+
+        # Specify the other timing information
+        duration = trs * tr_duration
+        basis_funcs = int(np.floor(duration / period))  # How bases do you have
+
+        if basis_funcs == 0:
+            err_msg = 'Too few timepoints (' + str(trs) + ') to accurately ' \
+                                                          'model drift'
+            logger.warning(err_msg)
+            basis_funcs = 1
+
+        noise_drift = np.zeros((timepoints.shape[0], basis_funcs))
+        for basis_counter in list(range(1, basis_funcs + 1)):
+
+            # What steps do you want to take for this basis function
+            timepoints_basis = (timepoints/basis_counter) + (np.random.rand()
+                                                             * np.pi * 2)
+
+            # Store the drift from this basis func
+            noise_drift[:, basis_counter - 1] = np.cos(timepoints_basis)
+
+        # Average the drift
+        noise_drift = np.mean(noise_drift, 1)
```

Comment:
Suggestion for future improvement: It is possible that the power of low frequency drift is higher than higher frequency drift. There may be some literature quantifying this property of fMRI data. It may be possible to weight fluctuations at different frequencies according to such pattern if it is reported in the literature, instead of putting equal weight. I am not expert on this. But I remember learning that fMRI power spectrum looks like 1/f.

Reply:
Good suggestion, I will keep it in mind